LibreNMS/Permissions.php

```diff
@@ -207,7 +210,9 @@ public function getDevicePermissions()
     public function getPortPermissions()
     {
         if (is_null($this->portPermissions)) {
-            $this->portPermissions = DB::table('ports_perms')->get();
+            $this->portPermissions = DB::table('ports_perms')
+                ->union($this->getDeviceGroupPermissionsQuery((new Port)))
```

Comment:
@Jellyfrog I just realized that this is silly, you don't need to include the device group query for ports.  Ports permissions is only for explicit port permissions.  Device groups only have to do with device permissions.

That should simplify your code.

Reply:
I'm a bit 50/50 with this.  
It really depends on how we use port permission in the rest of the app.
If it's only explicit, it always need a "double check" (device + port permission check)  
With this way you don't need to "double check"

Reply:
No, the Permissions cache was designed to not include device checks in ports checks...

If you want to add a helper to check device then port permissions that is fine.  Hitting the ports table with the query join like that could be a disaster in large installs.

Reply:
Actually, I kinda did this because this:
https://github.com/librenms/librenms/blob/ce628399a942015a09156ab3956d83eafeb1ac9e/app/Models/User.php#L182-L190

Reply:
Sure, but I don't like changing the design of Permissions.  I'm fairly firm on this.